core/src/main/java/uk/gov/gchq/koryphe/util/IterableUtil.java

```diff
@@ -87,4 +91,35 @@ public static <T> Iterable<T> limit(final Iterable<T> iterable, final int start,
         return new LimitedIterable<>(iterable, start, end, truncate);
     }
 
+    public static int size(final Iterable<?> iterable) {
+        if (iterable instanceof Collection) {
+            return ((Collection<?>) iterable).size();
+        }
+        return (int) StreamSupport.stream(iterable.spliterator(), false).count();
```

Comment:
Are these implementations the same that Guava used? Just wondering from an efficiency perspective

Reply:
Functionally it is the same. See [this class](https://github.com/google/guava/blob/1b9849143b4a5e555ff40663340aaf656390a430/guava/src/com/google/common/collect/Iterables.java#L127) and then [this one](https://github.com/google/guava/blob/1b9849143b4a5e555ff40663340aaf656390a430/guava/src/com/google/common/collect/Iterators.java#L167) from Guava. Only difference is the stream method is more concise than using `hasNext`.